pkgs/applications/science/electronics/verilog/default.nix

```
@@ -29,11 +34,32 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ readline ncurses bzip2 zlib ];
 
+  # tests from .travis.yml
+  doCheck = true; # runs ``make check``
+  # most tests pass, but some that rely on exact text of floating-point numbers fail on aarch64.
+  doInstallCheck = !stdenv.isAarch64;
+  installCheckInputs = [ perl ];
+
+  installCheckPhase = ''
+    # copy tests to allow writing results
+    export TESTDIR=$(mktemp -d)
+    cp -r ${iverilog-test}/* $TESTDIR
+
+    pushd $TESTDIR
+
+    # Run & check tests
+    PATH=$out/bin:$PATH perl vvp_reg.pl
+    # Check the tests, will error if unexpected tests fail. Some failures MIGHT be normal.
+    diff regression_report-devel.txt regression_report.txt
+    PATH=$out/bin:$PATH perl vpi_reg.pl
+
+    popd
+  '';
+
   meta = with stdenv.lib; {
     description = "Icarus Verilog compiler";
-    repositories.git = "https://github.com/steveicarus/iverilog.git";
     homepage = "http://iverilog.icarus.com/";
-    license = licenses.lgpl21;
+    license = licenses.lgpl21Plus;
```

```suggestion
    license = with licenses; [ gpl2Plus lgpl21Plus ] ;
```
I found only one mention of LGPL in tgt-vlog95/logic_lpm.c used for auto-generated files. The iverilog itself seems to be GPL2+.

---

Thanks for the catch, I guess I didn't look closely enough.